spark-operator-api/src/main/java/org/apache/spark/kubernetes/operator/status/ApplicationStateSummary.java

```diff
@@ -0,0 +1,121 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package org.apache.spark.kubernetes.operator.status;
+
+import java.util.Set;
+
+public enum ApplicationStateSummary implements BaseStateSummary {
+  /** Spark application is submitted to the cluster but yet scheduled. */
+  SUBMITTED,
+
+  /** Spark application will be restarted with same configuration */
+  SCHEDULED_TO_RESTART,
+
+  /** A request has been made to start driver pod in the cluster */
+  DRIVER_REQUESTED,
+
+  /** Driver pod has reached running state */
```

Comment:
To be clear, please use the matched K8s Pod Phase term exactly by using `Running` instead of `running`.
- https://kubernetes.io/docs/concepts/workloads/pods/pod-lifecycle/#pod-phase
```
- /** Driver pod has reached running state */
+ /** Driver pod has reached `Running` phase */
```